src/FakeItEasy/Creation/CastleDynamicProxy/CastleDynamicProxyGenerator.cs

```diff
@@ -115,12 +115,6 @@ internal static class CastleDynamicProxyGenerator
 
         public static bool CanGenerateProxy(Type typeOfProxy, [NotNullWhen(false)] out string? failReason)
         {
-            if (typeOfProxy.IsValueType)
-            {
-                failReason = DynamicProxyMessages.ProxyIsValueType(typeOfProxy);
-                return false;
-            }
-

```

Comment:
Are we certain we can never reach this point if `typeOfProxy` is a value type?

Reply:
I was last night! Let's recheck.

`CastleDynamicProxyGenerator.CanGenerateProxy` is called from

* `CastleDynamicProxyGenerator.GenerateClassProxy` is called from
    * `FakeObjectCreator.DefaultCreationStrategy.GenerateProxy` is called from
        * `FakeObjectCreator.DefaultCreationStrategy.TryCreateFakeWithDummyArgumentsForConstructor` is called from
            * `FakeObjectCreator.DefaultCreationStrategy.CreateFake`, which see one line below!
        *  `FakeObjectCreator.DefaultCreationStrategy.CreateFake` which see one line below!
* `FakeObjectCreator.DefaultCreationStrategy.CreateFake` is called from
    * `FakeObjectCreate.CreateFakeWithoutLoopDetection` is called from
        * `DummyValueResolver.ResolveByCreatingFakeStrategy.TryCreateDummyValue` _but only after `ResolveByActivatingValueTypeStrategy` did not create a Fake_. 🤔 I see now that we guard against the type of the Dummy being `void` there. I'm not even sure why we have that, nor how it might come up. So there might be a hole.
        * `FakeObjectCreator.CreateFake` is called from
            * `FakeAndDummyManager.CreateFake`, which has the new guard against trying to fake value types

As you can see, I am no longer so sure. The `ResolveByActivatingValueTypeStrategy` check (where we see that the Dummy we're trying to make isn't `void`) is something I missed last night. It's possible that this will only come up when we're making a default return value for a void-returning method, but I'm not 100% sure. I will investigate more. An easy workaround would be to add a `VoidDummyFactory` not unlike the `StringDummyFactory` and then we wouldn't need the special case.

(Also, I specifically made this change a separate commit so we could drop it if it proved problematic. I would be okay with that.)

Reply:
> I see now that we guard against the type of the Dummy being void there. I'm not even sure why we have that, nor how it might come up.

Like this:
```csharp
Create.Dummy(typeof(void));
```
But `A.Dummy<void>()` wouldn't work.

> An easy workaround would be to add a VoidDummyFactory

What would it do exactly? We can never create an instance of `void`... we should just fail to generate a dummy (which already happens).

As far as I can tell, we're good to go.